lib/DiscoveryStreamFeed.jsm

```diff
@@ -541,10 +545,16 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     if (this.isExpired({cachedData, key: "feed", url: feedUrl, isStartup})) {
       const feedResponse = await this.fetchFromEndpoint(feedUrl);
       if (feedResponse) {
+        const scoredItems = this.scoreItems(feedResponse.recommendations);
+        const {recsExpireTime} = feedResponse.settings;
+        const recommendations = this.rotate(scoredItems, recsExpireTime);
         this.componentFeedFetched = true;
         feed = {
           lastUpdated: Date.now(),
-          data: this.rotate(feedResponse),
```

Comment:
Rotate was unpacking the response, making changes to data, and repacking it.

I changed it so rotate takes the items, rotates them, and returns them. I then repackage it after calling it.

this was needed because I'm needing to call scoreItems now for both spocs and recs, but I only want to do rotate on recs, and rotate wasn't clumsy, I would have needed to make score items also unpack and repack, causing a bunch of pointless packing of the object.

This also meant I needed to make some changes to the way spocs call scoreItems.